### PR TITLE
PYTHON-3773 add types to client_session.py

### PR DESCRIPTION
**Pull request by sleepyStick**
https://jira.mongodb.org/browse/PYTHON-3773

### CODE REVIEW

File: pymongo/client_session.py [hunk 1]
```diff
@@ -851,7 +871,7 @@ def _finish_transaction(self, sock_info, command_name):
             sock_info, cmd, session=self, write_concern=wc, parse_write_concern_error=True
         )
 
-    def _advance_cluster_time(self, cluster_time):
+    def _advance_cluster_time(self, cluster_time: Mapping[str, Any]) -> None:
```
Comment (ShaneHarvey) [inline pymongo/client_session.py:874]: cluster_time is can also be None here.

File: pymongo/client_session.py [hunk 2]
```diff
@@ -873,7 +893,7 @@ def advance_cluster_time(self, cluster_time: Mapping[str, Any]) -> None:
             raise ValueError("Invalid cluster_time")
         self._advance_cluster_time(cluster_time)
 
-    def _advance_operation_time(self, operation_time):
+    def _advance_operation_time(self, operation_time: Timestamp) -> None:
```
Comment (ShaneHarvey) [inline pymongo/client_session.py:896]: operation_time can also be None here.

File: pymongo/client_session.py [hunk 3]
```diff
@@ -893,10 +913,10 @@ def advance_operation_time(self, operation_time: Timestamp) -> None:
             raise TypeError("operation_time must be an instance of bson.timestamp.Timestamp")
         self._advance_operation_time(operation_time)
 
-    def _process_response(self, reply):
+    def _process_response(self, reply: Mapping[str, Any]) -> None:
         """Process a response to a command that was run with this session."""
-        self._advance_cluster_time(reply.get("$clusterTime"))
-        self._advance_operation_time(reply.get("operationTime"))
+        self._advance_cluster_time(cast(Mapping[str, Any], reply.get("$clusterTime")))
+        self._advance_operation_time(cast(Timestamp, reply.get("operationTime")))
```
Comment (ShaneHarvey) [inline pymongo/client_session.py:919]: These casts should not be needed after the changes above.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/client_session.py (hunk 1 above)
Comment (sleepyStick) [reply pymongo/client_session.py]: I see, thanks for catching that! Fixed.

File: pymongo/client_session.py (hunk 2 above)
Comment (sleepyStick) [reply pymongo/client_session.py]: Done.

File: pymongo/client_session.py (hunk 3 above)
Comment (sleepyStick) [reply pymongo/client_session.py]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**